controllers/solrprometheusexporter_controller_test.go

```diff
@@ -200,6 +200,7 @@ func TestMetricsReconcileWithExporterConfig(t *testing.T) {
 	testMapsEqual(t, "deployment labels", util.MergeLabelsOrAnnotations(expectedDeploymentLabels, testDeploymentLabels), deployment.Labels)
 	testMapsEqual(t, "deployment annotations", testDeploymentAnnotations, deployment.Annotations)
 	testMapsEqual(t, "pod labels", util.MergeLabelsOrAnnotations(expectedDeploymentLabels, testPodLabels), deployment.Spec.Template.ObjectMeta.Labels)
+	testPodAnnotations["solr.apache.org/exporterConfigXmlMd5"] = "2c354e0aeb176a6b5560f46fe35c9ca7"
```

Comment:
Could you possibly do this dynamically, like: https://github.com/apache/lucene-solr-operator/pull/188/files#diff-c7a739f50a172286ab35c4729f5e1795249b1be34bd386695c9facb05d437fb4R274

I moved the `expectConfigMap` to be above the statefulSet check, so that I could use the data that was already fetched to make the Md5.